error: pattern matching in instanceof is not supported in -source 8

### Description

Cannot build project with react-native-svg 15.4.0
I am getting this error
`> Task :react-native-svg:compileReleaseJavaWithJavac FAILED
.../node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/FilterView.java:83: error: pattern matching in instanceof is not supported in -source 8
      if (node instanceof FilterPrimitiveView currentFilter) {
                                              ^
  (use -source 16 or higher to enable pattern matching in instanceof)
1 error
`

### Steps to reproduce

yarn add react-native-svg 15.4.0
build project
project won't compile

### Snack or a link to a repository

not needed

### SVG version

15.4.0

### React Native version

0.71.14

### Platforms

macOS

### JavaScript runtime

Hermes

### Workflow

React Native

### Architecture

None

### Build type

Release app & production bundle

### Device

None

### Device model

_No response_

### Acknowledgements

Yes

@bohdanprog I create an example like this 
npx react-native init example --version 0.70
yarn add react-native-svg
yarn add react-native-svg-transformer
add declaration.d.js 
`declare module "*.svg" {
  import { SvgProps } from "react-native-svg";
  const content: React.FC<SvgProps>;
  export default content;
}`
modify metro.config.js
`/**
 * Metro configuration for React Native
 * https://github.com/facebook/react-native
 *
 * @format
 */

const {getDefaultConfig} = require('metro-config');

module.exports = (async () => {
  const {
    resolver: {sourceExts, assetExts},
  } = await getDefaultConfig();
  return {
    transformer: {
      babelTransformerPath: require.resolve('react-native-svg-transformer'),
      getTransformOptions: async () => ({
        transform: {
          experimentalImportSupport: false,
          inlineRequires: true,
        },
      }),
    },
    resolver: {
      assetExts: assetExts.filter(ext => ext !== 'svg'),
      sourceExts: [...sourceExts, 'svg'],
    },
  };
})();`
and then run yarn android
* java version: 19.0.2
* gradle version: 7.5.1
* "react-native": "0.70.15",
* "react-native-svg": "^15.4.0",
* "react-native-svg-transformer": "^1.5.0"
* running in window
* android device: samsung galaxy s20

The problem reports:
node_modules\react-native-svg\android\src\main\java\com\horcrux\svg\FilterView.java:83: error: pattern matching in instanceof is not supported in -source 8
      if (node instanceof FilterPrimitiveView currentFilter) {
                                              ^
  (use -source 16 or higher to enable pattern matching in instanceof)

--- a/android/src/main/java/com/horcrux/svg/FilterView.java
+++ b/android/src/main/java/com/horcrux/svg/FilterView.java
@@ -80,7 +80,8 @@ class FilterView extends DefinitionView {
 
     for (int i = 0; i < getChildCount(); i++) {
       View node = getChildAt(i);
-      if (node instanceof FilterPrimitiveView currentFilter) {
+      if (node instanceof FilterPrimitiveView) {
+        FilterPrimitiveView currentFilter = (FilterPrimitiveView) node;
         res = currentFilter.applyFilter(mResultsMap, res);
         String resultName = currentFilter.getResult();
         if (resultName != null) {

